### PR TITLE
6: Remove grey text from gdocs

### PR DESCRIPTION
**Pull request by mruwnik**
https://github.com/StampyAI/GDocsRelatedThings/issues/6

### CODE REVIEW

File: parser/parser.js [hunk 1]
```diff
@@ -214,8 +214,17 @@ export const parseParagraph = (documentContext) => (paragraph) => {
   }
 };
 
+const isGrey = (textStyle) => {
+  const rgb = textStyle?.foregroundColor?.color?.rgbColor;
+  const rgbVals = rgb && Object.values(rgb);
+  const tolerance = 0.01;
+  return (
+    rgbVals && rgbVals.every((val) => Math.abs(val - rgbVals[0]) <= tolerance)
```
Comment (Aprillion) [inline parser/parser.js:222]: aah, let's not exclude explicitly-set-color black text... 🖤 

looking at Google Docs color picker, the first 2 options (0% and 26% brightness) look dark enough to keep as "normal" text IMHO, I would only exclude the other options (third one is 40% => anyting above 0.3 can be excluded .. but probably check with @plexish or with all editors on Discord)

Screenshot 2023-04-26 at 16 13 24

**Review by Aprillion — CHANGES_REQUESTED** [review]

File: parser/parser.js [hunk 2]
```diff
@@ -214,8 +214,17 @@ export const parseParagraph = (documentContext) => (paragraph) => {
   }
 };
 
+const isGrey = (textStyle) => {
+  const rgb = textStyle?.foregroundColor?.color?.rgbColor;
+  const rgbVals = rgb && Object.values(rgb);
+  const tolerance = 0.01;
```
Comment (Aprillion) [inline parser/parser.js:220]: 🤔 do we want to be this fancy? default gray colors in GDocs don't need "tolerance" between RGB values, `r === g && r === b && r >= 0.3` might be a more readable condition maybe..

Comment (mruwnik) [reply parser/parser.js:220]: they're floats, and I don't trust them to be equal :/ Though I suppose wrapping them with a round and toString would work. Or am I overthinking it?

Comment (Aprillion) [reply parser/parser.js:221]: I've never seen color channels as floats before, so no idea what to expect 😁

Comment (mruwnik) [reply parser/parser.js:221]: yup, they normalise them to be between 0-1, for some reason :D

File: parser/__tests__/parser.test.js [hunk 3]
```diff
@@ -72,6 +72,44 @@ describe("parsetextRun", () => {
...
+      };
+      expect(parsetextRun(textRun)).toBe("");
+    }
+  );
+
+  it.each([
+    // really dark grey or explicit black isn't removed
+    [{ red: 0.1, green: 0.1, blue: 0.1 }],
+    [{ red: 0.0, green: 0.0, blue: 0.0 }],
+    // really light grey or white isn't removed
```
Comment (Aprillion) [inline parser/__tests__/parser.test.js:96]: huh? why not?

Comment (mruwnik) [reply parser/__tests__/parser.test.js:96]: coz it's hard to differentiate from black. That being said, the cutoffs were too limiting, so I changed it to (0.02; 0.98) - that should suffice

Comment (mruwnik) [reply parser/__tests__/parser.test.js:96]: this handles all the available greys in gdocs, at least the ones that can currently be selected

Comment (Aprillion) [reply parser/__tests__/parser.test.js:96]: I mean, why not exclude white text?!?

File: parser/__tests__/parser.test.js [hunk 4]
```diff
@@ -72,6 +72,44 @@ describe("parsetextRun", () => {
...
+  );
+
+  it.each([
+    // really dark grey or explicit black isn't removed
+    [{ red: 0.01, green: 0.01, blue: 0.01 }],
+    [{ red: 0.0, green: 0.0, blue: 0.0 }],
+    // really light grey or white isn't removed
+    [{ red: 0.99, green: 0.99, blue: 0.99 }],
+    [{ red: 1.0, green: 1.0, blue: 1.0 }],
+    // non gray is removed
```
Comment (Aprillion) [inline parser/__tests__/parser.test.js:99]: non gray is *not* removed..

**Review by Aprillion — APPROVED** [review]
not sure why we want to include white text, but whatever, it's good enough to merge :D

File: parser/__tests__/parser.test.js (hunk 3 above)
Comment (mruwnik) [reply parser/__tests__/parser.test.js:96]: ah. That :D I assumed that if someone wants to have explicity white text for some obscure reason, then who am I to disallow it :D

File: parser/parser.js (hunk 1 above)
Comment (plexish) [reply parser/parser.js]: seems reasonable

**Pull request merged**